From: Ramon Novoa <ramon.novoa@artica.es>
Description: Bugfix by in braa_EncodeShortBitstringOid (32 bit numbers in OIDs)
Origin: https://github.com/mteg/braa/commit/7e1cab5e3fc359e91f4b6b3163ac795186032082 

--- a/braaasn.c
+++ b/braaasn.c
@@ -56,7 +56,7 @@
 		buffer[2] = (v & 0x7f);
 		return(3);
 	}
-	else
+	else if(! (v & 0xf0000000))
 	{
 		buffer[0] = ((v >> 21) & 0x7f) | 0x80;
 		buffer[1] = ((v >> 14) & 0x7f) | 0x80;
@@ -64,6 +64,15 @@
 		buffer[3] = (v & 0x7f);
 		return(4);
 	}
+	else
+	{
+		buffer[0] = ((v >> 28) & 0x7f) | 0x80;
+		buffer[1] = ((v >> 21) & 0x7f) | 0x80;
+		buffer[2] = ((v >> 14) & 0x7f) | 0x80;
+		buffer[3] = ((v >> 7) & 0x7f) | 0x80;
+		buffer[4] = (v & 0x7f);
+		return(5);
+	}
 }
 
 
@@ -72,7 +81,7 @@
 	u_int32_t ret = 0;
 	int i;
 	
-	for(i = 0; i < 4 && i < max; i++)
+	for(i = 0; i < 5 && i < max; i++)
 	{
 		ret <<= 7;
 		ret |= data[i] & 0x7f;
--- a/braaoids.c
+++ b/braaoids.c
@@ -29,10 +29,10 @@
 	buffer[0] = 0;
 	for(i = 0; i<o->len; i++)
 	{
-		unsigned char ib[12];
+		unsigned char ib[14];
 		int n;
 		
-		n = snprintf(ib, 10, ".%d", o->oid[i]);
+		n = snprintf(ib, 12, ".%d", o->oid[i]);
 		if((l + n + 1) < buffer_len)
 			sprintf(buffer, "%s%s", buffer, ib);
 		else
